### ISSUE TITLE
createSharedComposable never dispose when using SSR

### ISSUE THREAD

**Issue by zkulbeda** [vueuse/vueuse#3202]
### Describe the bug

A composable created by `createSharedComposable` is always reused in SSR between different requests due to [disabled vue lifecycle hooks in SSR](https://vuejs.org/guide/scaling-up/ssr.html#component-lifecycle-hooks:~:text=However%2C%20because%20the%20unmount%20hooks%20will%20never%20be%20called%20during%20SSR). I suppose it should create a composable scoped to vue root component and shared between its child components. Or docs should mention that this utility isn't ssr-frendly .

### Reproduction

[link](https://stackblitz.com/edit/github-ybprgx?file=app.vue)

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.20.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.2 - /usr/local/bin/npm
    pnpm: 8.6.3 - /usr/local/bin/pnpm
  npmPackages:
    @vueuse/core: ^10.2.1 => 10.2.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by bgondy** [comment]
I'm trying to create a composable that returns the same `ComputedRef`. This composable aims to be called tens of times per page.

For the record, I'm using Nuxt with SSR.

```ts
// composables/useIsAuthenticated.ts
import { createSharedComposable } from '@vueuse/core';

const useIsAuthenticated = createSharedComposable(() =>
  computed<boolean>(() => {
    // … 

    return false; // Always returning false for the example
  }),
);

export default useIsAuthenticated;
```

When doing this, the composable seems to be shared for every request on the server, although it works as expected in the browser, causing content mismatches.

IMHO, a warning message should be added to the documentation about this use case. WDYT ?

**Comment by romansp** [comment]
To those who are subscribed to this issue I believe it should be now fixed with #4997.
